### PR TITLE
Assorted small enhancements for node restart w/ pos

### PR DESCRIPTION
**Pull request by lazynina**

### CODE REVIEW

**Comment by lazynina** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1079?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* **#1079** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1079?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1073** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1073?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1072** <a href="https://app.graphite.dev/github/pr/deso-protocol/core/1072?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `feature/pos-networking-and-syncing`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1079?utm_source=stack-comment).

File: lib/miner.go [hunk 1]
```diff
@@ -319,8 +319,12 @@ func (desoMiner *DeSoMiner) Start() {
 			"start the miner")
 		return
 	}
-	glog.Infof("DeSoMiner.Start: Starting miner with difficulty target %s", desoMiner.params.MinDifficultyTargetHex)
 	blockTip := desoMiner.BlockProducer.chain.blockTip()
+	if desoMiner.params.IsPoSBlockHeight(blockTip.Header.Height) {
+		glog.Infof("DeSoMiner.Start: NOT starting miner because we are at a PoS block height %d", blockTip.Header.Height)
```
Comment (lazynina) [inline lib/miner.go:324]: No need to start the miner if we're running PoS!

File: lib/server.go [hunk 2]
```diff
@@ -496,13 +496,25 @@ func NewServer(
...
 	}
 
+	var headerCumWorkStr *string
+	headerCumWork := BigintToHash(_chain.headerTip().CumWork)
+	if headerCumWork != nil {
+		headerCumWorkString := hex.EncodeToString(headerCumWork[:])
+		headerCumWorkStr = &headerCumWorkString
+	}
+	var blockCumWorkStr *string
+	blockCumWork := BigintToHash(_chain.blockTip().CumWork)
```
Comment (lazynina) [inline lib/server.go:506]: same as above^

**Review by tholonious — APPROVED** [review]

**Comment by lazynina** [comment]
### Merge activity

* **Feb 29, 9:32 AM EST**: @lazynina started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/deso-protocol/core/1079).

**Pull request merged**